### ISSUE TITLE
diffInBusinessHours not working after upgrade to laravel 11

### ISSUE THREAD

**Issue by ziming** [kylekatarnls/business-time#71]
After i upgrade to laravel 11, I got this error from sentry

`Method diffInBusinesshours does not exist.`

Yes I notice the error from sentry put hours as `hours` instead of Hours

However, my code does not get it wrong

![CleanShot 2024-03-29 at 19 29 43@2x](https://github.com/kylekatarnls/business-time/assets/679513/ae872cf4-f39e-4492-afd2-becd83b823d3)

Before my upgrade to laravel 11, it is working. So I'm assuming this is an issue with Carbon 3 as I assume laravel 11 use Carbon 3 instead of Carbon 2

**Comment by kylekatarnls** [comment]
It could be but unit tests in this project are all ran with Carbon 3 for a while now.

**Comment by kylekatarnls** [comment]
Can you check `$this->created_at::class` and see if `business-time` is indeed applied to those classes.

By default, the mixin applies to:
```php
Carbon\Carbon
Carbon\CarbonImmutable
Illuminate\Support\Carbon
```

**Comment by ziming** [comment]
Hi,

Hope it helps

![CleanShot 2024-03-29 at 21 27 21@2x](https://github.com/kylekatarnls/business-time/assets/679513/4fdbe036-c3d0-4123-8780-85c8864bebf0)

**Comment by kylekatarnls** [comment]
Thank you for your help, could you also check that the service provider is still running and that it actually enabled the mixin, i.e. checking that this code is executed at boostrap:
https://github.com/kylekatarnls/business-time/blob/master/src/BusinessTime/Laravel/ServiceProvider.php#L27

**Comment by ziming** [comment]
Hmm not too sure what you mean but this is my BusinessTime code in my AppServiceProvider boot() method

![CleanShot 2024-03-29 at 21 31 20@2x](https://github.com/kylekatarnls/business-time/assets/679513/509a128e-ff61-470e-b60b-8266c2e18022)

**Comment by ziming** [comment]
let me know if that helped to answer ur question, thank you

**Comment by kylekatarnls** [comment]
Thank you, a last thing that could help to reproduce would be the stack trace of your error.

You may also try to upgrade nesbot/carbon, cmixin/business-day and cmixin/business-time to the latest version to be sure.

**Comment by ziming** [comment]
![CleanShot 2024-04-02 at 02 04 55@2x](https://github.com/kylekatarnls/business-time/assets/679513/394924e2-78da-4802-b53a-587ba6f8d672)

Stack trace attached. carbon, business day and business time are at their latest version too

**Comment by kylekatarnls** [comment]
I'm not yet able to reproduce. I installed a fresh Laravel 11 app and added the same `BusinessTime::enable(...)` in my `AppServiceProvider boot()` and I'm able to get a number out of `Carbon::parse('2024-05-01')->diffInBusinessHours('2024-05-01 12:00')`

I'm wondering if the boot order changed in Laravel 11, maybe this part of your code where you call `->created_at->diffInBusinessHours()` is executed before the application actually get booted?

**Comment by ziming** [comment]
Hmm i don't think so, my use of it is in a filament admin dashboard class where i show some stats

r u using Carbon\Carbon or Illuminate\Support\Carbon?

**Comment by kylekatarnls** [comment]
I tested both in the boot and both in the controller, they can be swap.

**Comment by kylekatarnls** [comment]
Hello, good news, it should be fixed by upgrading nesbot/carbon to 3.2.4.

**Comment by ziming** [comment]
what caused it? just curious

**Comment by kylekatarnls** [comment]
A variable mutation normalizing `diffIn<SomeThing>` into `diffIn<Something>` so it searched `diffInBusinesshours` instead of `diffInBusinessHours` and I struggled to find the case difference.

The fix in 3.2.4 is to this step in `__call` in isolation so normalization made for this step does not affect the rest (if it does not match a valid unit, it will fallback to search in mixin/macro with the method name not altered).

The `__call` magic method in Carbon is doing way too much, I will continue to decouple to minimize those kind of risks.
